hydrator-common/src/main/java/co/cask/hydrator/common/macro/MacroContext.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.common.macro;
+
+import java.util.Map;
+import java.util.TimeZone;
+import javax.annotation.Nullable;
+
+/**
+ * Context for macro substitution.
+ */
+public interface MacroContext {
+
+  long getLogicalStartTime();
+
+  @Nullable
+  TimeZone getTimeZone();
```

Comment:
The context is meant to supply any information a macro may need.  It can ignore this completely.